### PR TITLE
Fix binary cross entropy loss

### PR DESCRIPTION
**Pull request by VRehnberg**
Small fixes for errors that are thrown when not running on MPS. @oliveradk or @ejnnr, if you could check:
 - that it still runs on MPS
 - that it was logits you were expecting to be the output of the model

### CODE REVIEW

**Comment by ejnnr** [comment]
Thanks, tests run on MPS and I agree we want logits!

**Comment by oliveradk** [comment]
Thanks for catching the bce_w_logits bug and merging!
